qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -99,6 +99,13 @@ def save_page(self):
         """Save the current page."""
         self._widget.triggerPageAction(QWebEnginePage.SavePage)
 
+    def show_source(self, dispatcher, url):
+        # use view-source: scheme to show page source for webengine
+        url = QUrl('view-source:{}'.format(url.toString()))
+        new_tab = dispatcher._tabbed_browser.tabopen(
```

Comment:
Passing the `dispatcher` here creates a [coupling](https://en.wikipedia.org/wiki/Coupling_%28computer_programming%29) between two completely different parts of codes, which I'd rather avoid. Also, you're then accessing private variables on the dispatcher.

Instead, pass the `win_id` to the `*Action` objects when creating them, and get the tabbed-browser object using `tabbed_browser = objreg.get('tabbed-browser', scope='window', window=win_id)`.

Reply:
OK, nice!
What about the "dispatcher._current_widget()" in webkittab's show_source() method ?

Reply:
Ahh, found the answer:

widget = self._tabbed_browser.currentWidget()

So I can use the currentWidget() method of the tabbed_browser I get with objreg... nice!